src/Parallel/Main.hpp

```diff
@@ -639,39 +649,67 @@ void Main<Metavariables>::
 
 template <typename Metavariables>
 void Main<Metavariables>::execute_next_phase() {
-  if (Parallel::Phase::Exit == current_phase_) {
-    ERROR("Current phase is Exit, but program did not exit!");
-  }
-
-  const auto next_phase = PhaseControl::arbitrate_phase_change(
-      make_not_null(&phase_change_decision_data_), current_phase_,
-      *Parallel::local_branch(global_cache_proxy_));
-  if (next_phase.has_value()) {
-    // Only print info if there was an actual phase change.
-    if (current_phase_ != next_phase.value()) {
-      Parallel::printf("Entering phase from phase control: %s\n",
-                       next_phase.value());
-      current_phase_ = next_phase.value();
+  if (not exception_messages_.empty()) {
+    if (current_phase_ == Parallel::Phase::PostFailureCleanup) {
+      Parallel::printf(
+          "Received termination while cleaning up a previous termination. This "
+          "is cyclic behavior and cannot be supported. Cleanup must exit "
+          "cleanly without errors.");
+      sys::abort("");
     }
+    current_phase_ = Parallel::Phase::PostFailureCleanup;
+    Parallel::printf("Entering phase: %s\n", current_phase_)
```

Comment:
Missing semicolon. I forgot that in my other comment

Reply:
RIP :( fixed...